yal/askaquestion.py

```diff
@@ -502,16 +516,57 @@ async def state_handle_timeout_response(self):
         error, fields = await rapidpro.get_profile(whatsapp_id)
         if error:
             return await self.go_to_state("state_error")
-        data = {
-            "feedback_survey_sent": "",
-        }
+        self.save_metadata("feedback_timestamp", "")
+        data = {"feedback_survey_sent": "", "feedback_timestamp": ""}
         error = await rapidpro.update_profile(whatsapp_id, data)
         if error:
             return await self.go_to_state("state_error")
 
         timeout_type_sent = fields.get("feedback_type")
+        self.save_metadata("feedback_type", timeout_type_sent)
+
+        keyword = clean_inbound(self.inbound.content)
+        if keyword in self.AAQ_FEEDBACK_TRIGGER_KEYWORDS:
+            if timeout_type_sent == "ask_a_question":
+                return await self.go_to_state("state_handle_list_timeout")
+            if timeout_type_sent == "ask_a_question_2":
+                return await self.go_to_state("state_get_content_feedback")
+        else:
+            # Get it to display the message, instead of having this state try to
+            # match it to a keyword
+            self.inbound.session_event = Message.SESSION_EVENT.NEW
+            return await self.go_to_state("state_aaq_timeout_unrecognised_option")
 
+    async def state_aaq_timeout_unrecognised_option(self):
+        choices = [
+            Choice("feedback", self._("Reply to last text")),
+            Choice("mainmenu", self._("Go to the Main Menu")),
+            Choice("aaq", self._("Ask a question")),
+        ]
+        question = "\n".join(
+            [
+                "*[persona_emoji] Hmm, looks like you've run out of time to respond to "
+                "that message.*",
+                "",
+                "*What would you like to do now? Here are some options.*",
+                "",
+                get_display_choices(choices),
+            ]
+        )
+        timeout_type_sent = self.user.metadata.get("feedback_type")
         if timeout_type_sent == "ask_a_question":
-            return await self.go_to_state("state_handle_list_timeout")
+            feedback_state = "state_handle_list_timeout"
         if timeout_type_sent == "ask_a_question_2":
-            return await self.go_to_state("state_get_content_feedback")
+            feedback_state = "state_get_content_feedback"
+        return WhatsAppButtonState(
+            self,
+            question=question,
+            choices=choices,
+            error=get_generic_error(),
+            next={
+                "feedback": feedback_state,
+                # hardcode to prevent circular import
+                "mainmenu": "state_pre_mainmenu",
+                "aaq": self.START_STATE,
```

Comment:
We've been finding that this doesn't work in production. The user ends up getting sent to `state_start` instead of the state we want

Reply:
Ah of course, because we're using inheritance to split this flow into multiple flows (also a bit of a hack). I'll hardcode it